code/modules/projectiles/guns/launcher/crossbow.dm

```diff
@@ -237,16 +237,18 @@
 		draw(user)
 
 /obj/item/gun/launcher/crossbow/RCD/attackby(obj/item/W, mob/user)
-	if(istype(W, /obj/item/rcd_ammo))
-		if((stored_matter + 20) > max_stored_matter)
-			to_chat(user, "<span class='notice'>The RXD can't hold that many additional matter-units.</span>")
-			return
-		stored_matter += 20
-		qdel(W)
-		playsound(loc, 'sound/machines/click.ogg', 50, 1)
-		to_chat(user, "<span class='notice'>The RXD now holds [stored_matter]/[max_stored_matter] matter-units.</span>")
-		update_icon()
-		return
+	var/obj/item/stack/material/M = W
+	if(istype(M) && M.material.name == MATERIAL_COMPRESSED)
+		var/amount = min(M.get_amount(), round(max_stored_matter - stored_matter))
+		if(M.use(amount) && stored_matter < max_stored_matter)
+			stored_matter += amount
+			playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
+			to_chat(user, "<span class='notice'>You load [amount] Compressed Matter into \the [src].</span>. The RXD now holds [stored_matter]/60 matter-units.")
+			update_icon()	//Updates the ammo counter
+		if (M.use(amount) && stored_matter >= max_stored_matter)
+			to_chat(user, "<span class='notice'>The RXD is full.")
```

Comment:
You can replace these with span macros (`SPAN_NOTICE`)

Reply:
A bit too new on coding to know what that means, searched around the web for 'span macros' and came up empty. I just put that bit in line with the loading for matter projectiles, cleaned up that bit outside the span class bit, removed the alert for it being full.